registry-ui/src/hooks/useGenerateReportDialog.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+    Alert,
+    AlertTitle,
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    Stack,
+    Typography,
+    MenuItem,
+    FormControl,
+    Select,
+    InputLabel,
+    SelectChangeEvent
+} from "@mui/material";
+import { useQuery } from "@tanstack/react-query";
+
+import React, { useState } from "react";
+import { GenerateReportResponse } from "react-libs/provena-interfaces/ProvenanceAPI";
+import { useGenerateReport } from "./useGenerateReport";
+import { ItemSubType } from "provena-interfaces/RegistryModels";
+
+export interface GenerateReportProps {
+    nodeId: string | undefined;
+    itemSubType: ItemSubType | undefined; 
+    // This is a wrapper around the other child components and acts as as a callback. 
+    onSuccess?: (response: GenerateReportResponse) => void;
+}
+
+const DEPTH_LIST = ["1", "2", "3"]
+
+export const useGenerateReportDialog = (props: GenerateReportProps) => {
+
+    const [popUpOpen, setPopupOpen] = useState<boolean>(false);
+    const [depth, setDepth] = useState<string>("");
```

Comment:
If you need it to be a string for your query string, then handle this later - don't muddy up the client implementation so that it can be sent easier to the API.